apiserver/cache/redis.go

```diff
@@ -106,3 +108,32 @@ func (r *Cache) SaveJti(ctx goContext.Context, userId int, jti string, expiratio
 func (r *Cache) FlushAll(ctx goContext.Context) error {
 	return r.client.FlushAll(ctx).Err()
 }
+
+func testInvoiceRateLimiterKey(userId int) string {
+	return fmt.Sprintf("test:invoices:userId:%d", userId)
+}
+
+func (r *Cache) IncrementTestInvoiceCounter(ctx goContext.Context, userId int) (int, error) {
+	key := testInvoiceRateLimiterKey(userId)
+	var v int64
+
+	err := r.client.Watch(ctx, func(tx *redis.Tx) error {
+		var err error
+		v, err = tx.Incr(ctx, key).Result()
+		if err != nil {
+			return err
+		}
+		if v == 1 {
```

Comment:
add comment here too. Explain that when the key is just created we need to set proper expiration too